syncstorage/storage/sql/queries_generic.py

```diff
@@ -92,6 +93,46 @@
 DELETE_ITEMS = "DELETE FROM %(bso)s WHERE userid=:userid "\
                "AND collection=:collectionid AND id IN %(ids)s"
 
+CREATE_BATCH = "INSERT INTO batch_uploads (batch, userid, collection) "\
+                     "VALUES (:batch, :userid, :collection)"
+
+VALID_BATCH = "SELECT batch FROM batch_uploads WHERE batch = :batch " \
+                    "AND userid = :userid AND collection = :collection"
+
+APPEND_ITEMS_TO_BATCH = "INSERT INTO %(bui)s "\
+                              "  (batch, id, sortindex, modified, payload, "\
+                              "   payload_size, ttl) "\
+                              "VALUES "\
+                              "  (:batchid, :id, :sortindex, :modified, "\
+                              "   :payload, :payload_size, :ttl) "\
+                              "ON DUPLICATE KEY UPDATE "\
+                              "  sortindex = VALUES(sortindex), "\
+                              "  payload = VALUES(payload), "\
+                              "  payload_size = VALUES(payload_size) "\
+                              "  ttl = VALUES(ttl)"
+
+APPLY_BATCH = "INSERT INTO %(bso)s "\
+                     "  (userid, collection, id, sortindex, modified, "\
+                     "   payload, payload_size, ttl) "\
+                     "SELECT "\
+                     "  :userid, :collection, id, sortindex, :modified, " \
+                     "  COALESCE(payload, \"\"), COALESCE(payload_size, 0), "\
+                     "  COALESCE(ttl, ttl + :default_ttl) "\
```

Comment:
Since the `ttl` column as a default value, I suspect we can just select `ttl` here and let the default do the right thing, rather than trying to `COALESCE` it to a specific non-`NULL` value.